ui/raidboss/emulator/data/RaidEmulator.js

```diff
@@ -13,10 +14,14 @@ export default class RaidEmulator extends EventBus {
     this.lastLogTimestamp = null;
     this.internalTimestampTracker = null;
   }
+  setOptions(options) {
+    this.options = options;
+  }
   addEncounter(encounter) {
     this.encounters.push(encounter);
   }
   setCurrent(index) {
+    console.assert(this.options);
```

Comment:
Maybe this `console.assert` should have a message of some sort? Otherwise raidemulator-related changes look good.